tests/jballerina-integration-test/src/test/resources/transaction/src/participantservice/01_remote_participant.bal

```diff
@@ -284,7 +284,10 @@ function initiatorFunc(boolean throw1, boolean throw2,
         log:printInfo("aborted ran");
         S1 = S1 + " aborted-block";
     }
-    if (!isAbort && remote2) {
+    // Commit order of transaction initiator and participants cannot be guaranteed.
+    // Hence following logic will wait until participant function get committed.
+    // We need to skip cases like abort and retry without running remote participant second time.
+    if (!isAbort && !(trx_ran_once && !remote2)) {
```

Comment:
(!isAbort && !trx_ran_once && remote2) ==> committed && firstTime && remote2

Is it intended not to wait if remote2 is false? Maybe we don't have a test case for that.

Reply:
We cannot have a test with remote2 = true and without trx_ran_once. Anyway I'll add check for that path.